bokehjs/src/lib/models/axes/axis.ts

```diff
@@ -320,7 +327,15 @@ export abstract class AxisView extends GuideRendererView {
 
     const [nx, ny] = this.normals
     const orient = this.model.axis_label_orientation
-    const standoff = extents.tick + extents.tick_label + this.model.axis_label_standoff
+    const standoff_mode = this.model.axis_label_standoff_mode
+    const standoff = (() => {
+      switch (standoff_mode) {
+        case "tick_labels":
+          return extents.tick + extents.tick_label + this.model.axis_label_standoff
+        case "axis":
+          return this.model.axis_label_standoff
+      }
+    })()
```

Comment:
Given that `axis_label_standoff` is always included, then I would suggest, e.g.:

```suggestion
    const standoff = this.model.axis_label_standoff + (() => {
      switch (standoff_mode) {
        case "tick_labels":  return extents.tick + extents.tick_label
        case "axis":         return 0
      }
    })()
```
For readability you can split off the right hand side into a separate const value. Same in the other case(s).

Reply:
agreed, is updated now